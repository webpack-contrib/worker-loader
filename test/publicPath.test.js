import path from 'path';

import { customAlphabet } from 'nanoid';

import {
  compile,
  getCompiler,
  getErrors,
  getModuleSource,
  getResultFromBrowser,
  getWarnings,
} from './helpers';

describe('"publicPath" option', () => {
  it('should work and use "__webpack_public_path__" by default', async () => {
    const compiler = getCompiler('./chunks/entry.js');
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    expect(getModuleSource('./chunks/worker.js', stats)).toMatchSnapshot(
      'module'
    );
    expect(result).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work and respect the "publicPath" option ("string")', async () => {
    const compiler = getCompiler('./chunks/entry.js', {
      publicPath: '/public-path-static/',
    });
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    expect(getModuleSource('./chunks/worker.js', stats)).toMatchSnapshot(
      'module'
    );
    expect(result).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work and respect the "publicPath" option ("function")', async () => {
    const compiler = getCompiler('./chunks/entry.js', {
      publicPath: () => {
        return `/public-path-static/`;
      },
    });
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    expect(getModuleSource('./chunks/worker.js', stats)).toMatchSnapshot(
      'module'
    );
    expect(result).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work and respect the "output.publicPath" option default value', async () => {
    const nanoid = customAlphabet('1234567890abcdef', 10);
    const compiler = getCompiler(
      './chunks/entry.js',
      {},
      {
        output: {
          path: path.resolve(__dirname, './outputs', `test_${nanoid()}`),
          filename: '[name].bundle.js',
          chunkFilename: '[name].chunk.js',
          publicPath: '',
        },
      }
    );
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    expect(getModuleSource('./chunks/worker.js', stats)).toMatchSnapshot(
      'module'
    );
    expect(result).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work and respect the "output.publicPath" option value ("string")', async () => {
    const nanoid = customAlphabet('1234567890abcdef', 10);
    const compiler = getCompiler(
      './chunks/entry.js',
      {},
      {
        output: {
          publicPath: '/public-path-static/',
          path: path.resolve(__dirname, './outputs', `test_${nanoid()}`),
          filename: '[name].bundle.js',
          chunkFilename: '[name].chunk.js',
        },
      }
    );
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    expect(getModuleSource('./chunks/worker.js', stats)).toMatchSnapshot(
      'module'
    );
    expect(result).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work and respect the "output.publicPath" option value ("function")', async () => {
    const nanoid = customAlphabet('1234567890abcdef', 10);
    const compiler = getCompiler(
      './chunks/entry.js',
      {},
      {
        output: {
          publicPath: () => '/public-path-static/',
          path: path.resolve(__dirname, './outputs', `test_${nanoid()}`),
          filename: '[name].bundle.js',
          chunkFilename: '[name].chunk.js',
        },
      }
    );
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    expect(getModuleSource('./chunks/worker.js', stats)).toMatchSnapshot(
      'module'
    );
    expect(result).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work and respect "filename" and "chunkFilename" option values', async () => {
    const nanoid = customAlphabet('1234567890abcdef', 10);
    const compiler = getCompiler(
      './chunks/entry.js',
      {},
      {
        module: {
          rules: [
            {
              test: /worker\.js$/i,
              rules: [
                {
                  loader: path.resolve(__dirname, './../src'),
                },
              ],
            },
          ],
        },
        output: {
          publicPath: '/public-path-static-other/',
          path: path.resolve(__dirname, './outputs', `test_${nanoid()}`),
          filename: 'other-static/js/[name].bundle.js',
          chunkFilename: 'other-static/js/[name].chunk.js',
        },
      }
    );
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    expect(getModuleSource('./chunks/worker.js', stats)).toMatchSnapshot(
      'module'
    );
    expect(result).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work and respect "filename" and "chunkFilename" option values', async () => {
    const compiler = getCompiler('./chunks/entry.js', {
      publicPath: '/public-path-static-other/',
      filename: 'other-static/js/[name].worker.js',
      chunkFilename: 'other-static/js/[name].chunk.worker.js',
    });
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    expect(getModuleSource('./chunks/worker.js', stats)).toMatchSnapshot(
      'module'
    );
    expect(result).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });
});
