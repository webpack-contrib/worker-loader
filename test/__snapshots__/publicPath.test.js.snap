// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"publicPath" option should work and respect "filename" and "chunkFilename" option values: errors 1`] = `Array []`;

exports[`"publicPath" option should work and respect "filename" and "chunkFilename" option values: errors 2`] = `Array []`;

exports[`"publicPath" option should work and respect "filename" and "chunkFilename" option values: module 1`] = `
"export default function Worker_fn() {
  return new Worker(__webpack_public_path__ + \\"other-static/js/worker.bundle.worker.js\\");
}
"
`;

exports[`"publicPath" option should work and respect "filename" and "chunkFilename" option values: module 2`] = `
"export default function Worker_fn() {
  return new Worker(__webpack_public_path__ + \\"other-static/js/worker.worker.js\\");
}
"
`;

exports[`"publicPath" option should work and respect "filename" and "chunkFilename" option values: result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"publicPath" option should work and respect "filename" and "chunkFilename" option values: result 2`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"publicPath" option should work and respect "filename" and "chunkFilename" option values: warnings 1`] = `Array []`;

exports[`"publicPath" option should work and respect "filename" and "chunkFilename" option values: warnings 2`] = `Array []`;

exports[`"publicPath" option should work and respect the "output.publicPath" option default value: errors 1`] = `Array []`;

exports[`"publicPath" option should work and respect the "output.publicPath" option default value: module 1`] = `
"export default function Worker_fn() {
  return new Worker(__webpack_public_path__ + \\"test.worker.js\\");
}
"
`;

exports[`"publicPath" option should work and respect the "output.publicPath" option default value: result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"publicPath" option should work and respect the "output.publicPath" option default value: warnings 1`] = `Array []`;

exports[`"publicPath" option should work and respect the "output.publicPath" option value ("function"): errors 1`] = `Array []`;

exports[`"publicPath" option should work and respect the "output.publicPath" option value ("function"): module 1`] = `
"export default function Worker_fn() {
  return new Worker(__webpack_public_path__ + \\"test.worker.js\\");
}
"
`;

exports[`"publicPath" option should work and respect the "output.publicPath" option value ("function"): result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"publicPath" option should work and respect the "output.publicPath" option value ("function"): warnings 1`] = `Array []`;

exports[`"publicPath" option should work and respect the "output.publicPath" option value ("string"): errors 1`] = `Array []`;

exports[`"publicPath" option should work and respect the "output.publicPath" option value ("string"): module 1`] = `
"export default function Worker_fn() {
  return new Worker(__webpack_public_path__ + \\"test.worker.js\\");
}
"
`;

exports[`"publicPath" option should work and respect the "output.publicPath" option value ("string"): result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"publicPath" option should work and respect the "output.publicPath" option value ("string"): warnings 1`] = `Array []`;

exports[`"publicPath" option should work and respect the "publicPath" option ("function"): errors 1`] = `Array []`;

exports[`"publicPath" option should work and respect the "publicPath" option ("function"): module 1`] = `
"export default function Worker_fn() {
  return new Worker(__webpack_public_path__ + \\"test.worker.js\\");
}
"
`;

exports[`"publicPath" option should work and respect the "publicPath" option ("function"): result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"publicPath" option should work and respect the "publicPath" option ("function"): warnings 1`] = `Array []`;

exports[`"publicPath" option should work and respect the "publicPath" option ("string"): errors 1`] = `Array []`;

exports[`"publicPath" option should work and respect the "publicPath" option ("string"): module 1`] = `
"export default function Worker_fn() {
  return new Worker(__webpack_public_path__ + \\"test.worker.js\\");
}
"
`;

exports[`"publicPath" option should work and respect the "publicPath" option ("string"): result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"publicPath" option should work and respect the "publicPath" option ("string"): warnings 1`] = `Array []`;

exports[`"publicPath" option should work and use "__webpack_public_path__" by default: errors 1`] = `Array []`;

exports[`"publicPath" option should work and use "__webpack_public_path__" by default: module 1`] = `
"export default function Worker_fn() {
  return new Worker(__webpack_public_path__ + \\"test.worker.js\\");
}
"
`;

exports[`"publicPath" option should work and use "__webpack_public_path__" by default: result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"publicPath" option should work and use "__webpack_public_path__" by default: warnings 1`] = `Array []`;
